### PR TITLE
Implement changes for Upgrades to the inline config of packages

### LINKED ISSUE #3190: Upgrades to the inline config of packages should get reflected
**Bug description**
Upgrades to the inline config of packages should get reflected
As a user, I would like to be able to upgrade configurations of package with inline valuesFrom.

**Affected product area (please put an X in all that apply)**

- ( ) APIs
- (X) Addons
- (X) CLI
- ( ) Docs
- ( ) IAM
- ( ) Installation
- ( ) Plugin
- ( ) Security
- ( ) Test and Release
- ( ) User Experience
- ( ) Developer Experience

**Expected behavior**

**Steps to reproduce the bug**

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

### PR DESCRIPTION
**Pull request by shivaani0505**
### What this PR does / why we need it
Implement changes for Upgrades to the inline config of packages

### Which issue(s) this PR fixes

Fixes #3190 

### Release note

```release-note
Upgrades to the inline config of packages should get reflected
```

#### Special notes for your reviewer
**Testing:**
env tests pass for the changes

### CODE REVIEW

File: addons/controllers/clusterbootstrap_controller.go [hunk 1]
```diff
@@ -67,6 +67,7 @@ type ClusterBootstrapReconciler struct {
 	aggregatedAPIResourcesClient client.Client
 	// helper for looking up api-resources and getting preferred versions
 	gvrHelper util.GVRHelper
+	cbHelper  *clusterbootstrapclone.Helper
```
Comment (vijaykatam) [inline addons/controllers/clusterbootstrap_controller.go:70]: Lets not store this as a controller level variable that is long lived, if we need to we need to investigate for memory leaks in a scaled environment before we can merge this.

File: addons/controllers/clusterbootstrap_controller.go [hunk 2]
```diff
@@ -1149,6 +1150,12 @@ func (r *ClusterBootstrapReconciler) GetDataValueSecretNameFromBootstrapPackage(
 			if err = r.Get(r.context, key, secret); err != nil {
 				r.Log.Error(err, "unable to fetch secret for package with inline config", "objectkey", key)
 				return "", err
+			} else {
+				// secret for package with inline does not exist, we should create one
+				_, err := r.cbHelper.CreateSecretFromInline(cluster, cbPkg, packageRefName)
```
Comment (maralavi) [inline addons/controllers/clusterbootstrap_controller.go:1155]: @shivaani0505 Did you test this in a cluster and worked well? Why this change takes care of updating inline values?

An alternative would be to do inline processing in reconcileNormal right after the call to ```createOrPatchClusterBootstrapFromTemplate()``` (for both create and update).

File: addons/controllers/clusterbootstrap_controller.go (hunk 1 above)
Comment (shivaani0505) [reply addons/controllers/clusterbootstrap_controller.go:70]: I will take a look today at this one

File: addons/controllers/clusterbootstrap_controller.go (hunk 2 above)
Comment (shivaani0505) [reply addons/controllers/clusterbootstrap_controller.go]: Updated code to address the highlighted issue.

File: addons/controllers/clusterbootstrap_controller.go (hunk 1 above)
Comment (shivaani0505) [reply addons/controllers/clusterbootstrap_controller.go]: Fixed

File: addons/controllers/clusterbootstrap_controller.go [hunk 3]
```diff
@@ -1142,13 +1142,28 @@ func (r *ClusterBootstrapReconciler) GetDataValueSecretNameFromBootstrapPackage(
...
 		if cbPkg.ValuesFrom.Inline != nil {
 			packageSecretName := util.GeneratePackageSecretName(cluster.Name, packageRefName)
 			secret := &corev1.Secret{}
 			key := client.ObjectKey{Namespace: cluster.Namespace, Name: packageSecretName}
 			if err = r.Get(r.context, key, secret); err != nil {
-				r.Log.Error(err, "unable to fetch secret for package with inline config", "objectkey", key)
-				return "", err
+				if apierrors.IsNotFound(err) {
+					// secret for package with inline does not exist, we should create one
+					_, err := clusterBootstrapHelper.CreateSecretFromInline(cluster, cbPkg, packageRefName)
```
Comment (vijaykatam) [inline addons/controllers/clusterbootstrap_controller.go:1154]: we also need to ensure ownerRef of ClusterBootstrap object on the created secret similar to how it is done for create

File: addons/controllers/clusterbootstrap_controller.go [hunk 4]
```diff
@@ -1142,13 +1142,28 @@ func (r *ClusterBootstrapReconciler) GetDataValueSecretNameFromBootstrapPackage(
...
+					if err != nil {
+						return "", err
+					}
+				} else {
+					r.Log.Error(err, "unable to fetch secret for package with inline config", "objectkey", key)
+					return "", err
+				}
+			} else {
+				_, err = clusterBootstrapHelper.CreateOrPatchInlineSecret(cluster, cbPkg, secret)
```
Comment (vijaykatam) [inline addons/controllers/clusterbootstrap_controller.go:1163]: Same here, lets ensure ownerRef of ClusterBootstrap object

**Review by vijaykatam — COMMENTED** [review]
Lets add tests for updating inline secret to clusterbootstrap_controller_test with 2 scenarios.
1. Adding an inline value for an existing cb object.
2. Updating an inline value for an existing cb object.

File: addons/controllers/clusterbootstrap_controller.go (hunk 3 above)
Comment (shivaani0505) [reply addons/controllers/clusterbootstrap_controller.go]: Fixed

File: addons/controllers/clusterbootstrap_controller.go (hunk 4 above)
Comment (shivaani0505) [reply addons/controllers/clusterbootstrap_controller.go]: Fixed

**Review by vijaykatam — COMMENTED** [review]
Change looks good to me. Please add tests as requested in https://github.com/vmware-tanzu/tanzu-framework/pull/3191#pullrequestreview-1092515927 so that this change can be merged.

**Comment by shivaani0505** [comment]
> Change looks good to me. Please add tests as requested in [#3191 (review)](https://github.com/vmware-tanzu/tanzu-framework/pull/3191#pullrequestreview-1092515927) so that this change can be merged.

Test case has been added and I have verified with tkgs cluster
Test Logs:
```
Workload cluster 'cc-inline' created

STEP: Generating credentials for workload cluster "cc-inline"
Retrieving credentials for workload cluster cc-inline 
getting secret for cluster
waiting for resource cc-inline-kubeconfig of type *v1.Secret to be up and running
Merging credentials into kubeconfig file /Users/gshivaani/kubectl-vsphere-plugin/bin/testinline.config 
Credentials of cluster 'cc-inline' have been saved 
STEP: Get k8s client for workload cluster "cc-inline"
You can now access the cluster by running 'kubectl config use-context cc-inline-admin@cc-inline' under path '/var/folders/nd/q7bzmcrj5g70mdg92q3c7pb00000gp/T/cc-inline.kubeconfig' 
I0907 17:41:52.826848   95213 request.go:665] Waited for 1.030994962s due to client-side throttling, not priority and fairness, request: GET:https://192.168.116.2:6443/apis/stats.antrea.io/v1alpha1?timeout=32s
STEP: Update inline config for attribute periodSeconds for metrics server package for "cc-inline"

• [SLOW TEST:371.734 seconds]
TKGS ClusterClass based workload cluster tests
/Users/gshivaani/tanzu-framework/pkg/v1/tkg/test/tkgctl/tkgs_cc/tkgs_cc_workload_cluster_test.go:36
  when input file is cluster class based with custom Cluster Bootstrap
  /Users/gshivaani/tanzu-framework/pkg/v1/tkg/test/tkgctl/tkgs_cc/tkgs_cc_workload_cluster_test.go:118
    should create the data value secret in supervisor and guest cluster for a package with inline config
    /Users/gshivaani/tanzu-framework/pkg/v1/tkg/test/tkgctl/tkgs_cc/tkgs_cc_workload_cluster_test.go:181
------------------------------
SS
JUnit report was created: /var/folders/nd/q7bzmcrj5g70mdg92q3c7pb00000gp/T/artifacts/junit/junit.e2e_suite.1.xml

Ran 1 of 10 Specs in 372.798 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 9 Skipped
PASS | FOCUSED
```

**Review by vijaykatam — APPROVED** [review]
Looks good. Thanks for updating tests!
